### ISSUE TITLE
Game freezes when you search for a beatmap in the carousel with the search bar

### ISSUE THREAD

**Issue by Gabixel** [ppy/osu#11571]
**Describe the bug:** Not sure if it's related to my large number of beatmaps in the carousel, but something has changed recently which seems to reduce performance when searching for a beatmap in the carousel with the search bar.

**Screenshots or videos showing encountered issue:** 
- [`2021.118.1`](https://github.com/ppy/osu/releases/tag/2021.118.1)

https://user-images.githubusercontent.com/43073074/105505316-1b3b8700-5cc9-11eb-8eb0-d00f74a9a393.mp4

- previous version: [`2021.109.0`](https://github.com/ppy/osu/releases/tag/2021.109.0)

https://user-images.githubusercontent.com/43073074/105505635-7bcac400-5cc9-11eb-9972-dbc8796afaa9.mp4

**osu!lazer version:** `2021.118.1`

(The logs don't seem to give any information about it)

**Comment by bdach** [comment]
Do you mind uploading a video of the freeze with the performance details view (<kbd>Ctrl</kbd>+<kbd>F2</kbd>) visible and scrolled to end (so that the top 5 longest updating components are visible)?

**Comment by Gabixel** [comment]
here (the video begins with the first loading of the carousel, then I tried various searches):

https://user-images.githubusercontent.com/43073074/105522470-9313ac80-5cdd-11eb-9fec-4b7ab7ea8864.mp4

**Comment by bdach** [comment]
This seems to narrow it down to `BeatmapCarousel` - that said, that class had no changes to it between 2020.109.0 and 2020.118.1. There were a few changes in 2020.109.0 that could have changed performance, but seeing that it seems to work ok, they don't look relevant. Looks like further investigation will be needed on our behalf.

**Comment by peppy** [comment]
@Gabixel is this still reproducible for you? If possible could you please share your client.db? Will let me profile your specific use case locally.

**Comment by Gabixel** [comment]
Still looks reproducible.

[Here's a link to the `client.db` with the issue](https://drive.google.com/uc?export=download&id=1z7lRwDUI79Ui_KwWxaTFWqwDeZ3k1zvk).

**Comment by peppy** [comment]
Preliminary investigation shows that this was  caused by the switch to .NET 5, and potentially a different pathway being used for `IndexOf` for better unicode support (presuming that's what ICU stands for..?)

![20210209 175727 (Parallels Desktop app)](https://user-images.githubusercontent.com/191335/107339752-64306f80-6b00-11eb-923b-0b47cea9ab34.png)

**Comment by Zeno-ZX** [comment]
Still have this issue when searching with a large amount of beatmaps. https://youtu.be/Dt4jhfRwv4M

A friend who doesn't have a lot of beatmaps does **not** have this issue, and searching is smooth as expected.
If needed, I can grab an install with a small amount of maps and test.

[If you are seeing this immediately, YouTube may still be processing the video.]

**Comment by peppy** [comment]
As you may notice, this issue is still open, which means the issue has not been fixed yet.

**Comment by Zeno-ZX** [comment]
> As you may notice, this issue is still open, which means the issue has not been fixed yet.

Yes, I have noticed this. I suppose I should've changed my wording- I just wanted to see if this was still being addressed, and it seems to be. This is the one thing stopping me from switching to osu!lazer as my main client.
 
Thanks for everything ppy!
